src/wordle_solver/solver.py

```diff
@@ -22,69 +24,86 @@
     "m", "h", "g", "b", "f", "y", "w", "k", "v", "x", "z", "j", "q"
 ]
 
-def _get_guess(word_data: dict, guesses: int, words: WordList) -> str:
+
+def _get_guess(word_data: dict, guesses: int, words: WordList) -> WordList | str:
     """Given a word list, number of guesses and past guesses, return a new word."""
     if guesses == 0:
         return "orate"
-    
+
     # filter for letters we know and definitely don't know
-    reg_filter = ["[a-z]", "[a-z]", "[a-z]", "[a-z]", "[a-z]"]
+    reg_filter = ["", "", "", "", ""]
     for i in range(5):
         if i in word_data["correct"]:
             reg_filter[i] = word_data["correct"][i]
+        elif i in word_data["present_position"]:
+            reg_filter[
+                i] = f"[^{(word_data['absent'] - word_data['present']) | set(word_data['present_position'][i])} ]"
         else:
             reg_filter[i] = f"[^{word_data['absent'] - word_data['present']}]"
     reg_filter = re.compile("".join(reg_filter))
+    logging.info(f"reg_filter: {reg_filter}")
```

Comment:
Consider using `f"{reg_filter=}"` or `f"{reg_filter = }"`

https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging